core/src/pretty.rs

```diff
@@ -832,50 +851,49 @@ where
             FunPattern(pat, body) => allocator.function(allocator.pat_with_parens(pat), body),
             // Format this as the application `std.contract.from_predicate <pred>`.
             Lbl(_lbl) => allocator.text("%<label>").append(allocator.line()),
-            Let(id, rt, body, attrs) => docs![
+            Let(bindings, body, attrs) => docs![
                 allocator,
                 "let ",
                 if attrs.rec {
                     allocator.text("rec ")
                 } else {
                     allocator.nil()
                 },
-                id.to_string(),
-                if let Annotated(annot, _) = rt.as_ref() {
-                    annot.pretty(allocator)
-                } else {
-                    allocator.nil()
-                },
-                allocator.line(),
-                "= ",
-                if let Annotated(_, inner) = rt.as_ref() {
-                    inner.pretty(allocator)
-                } else {
-                    rt.pretty(allocator)
-                },
+                allocator.intersperse(
+                    bindings
+                        .iter()
+                        .map(|(k, v)| allocator.binding(*k, v.clone())),
+                    docs![allocator, ",", allocator.line()]
+                ),
                 allocator.line(),
                 "in",
             ]
             .nest(2)
             .append(allocator.line())
             .append(body.pretty(allocator).nest(2))
             .group(),
-            LetPattern(pattern, rt, body) => docs![
+            LetPattern(bindings, body) => docs![
                 allocator,
                 "let ",
-                pattern,
-                if let Annotated(annot, _) = rt.as_ref() {
-                    annot.pretty(allocator)
-                } else {
-                    allocator.nil()
-                },
-                allocator.line(),
-                "= ",
-                if let Annotated(_, inner) = rt.as_ref() {
-                    inner
-                } else {
-                    rt
-                },
+                allocator.intersperse(
+                    bindings.iter().map(|(pat, rt)| docs![
+                        allocator,
+                        pat,
+                        if let Annotated(annot, _) = rt.as_ref() {
+                            annot.pretty(allocator)
+                        } else {
+                            allocator.nil()
+                        },
+                        allocator.line(),
+                        "= ",
+                        if let Annotated(_, inner) = rt.as_ref() {
+                            inner
+                        } else {
+                            rt
+                        },
+                    ]),
+                    docs![allocator, ",", allocator.line()]
```

Comment:
Couldn't we factor this with `bindings`? For example by taking anything that implement `Pretty` as the first argument instead of `Ident`? It seems that the code pretty similar, just with `pat` instead of `id` (maybe we need to implement `Pretty` for `(Loc)Ident`, but that would make sense).

Reply:
Yes, that turns out nicely, thanks!